### PR TITLE
Format consumer config as table, add remaining fields

### PR DESCRIPTION
**Pull request by bruth**

### CODE REVIEW

**Comment by bruth** [comment]
Updated the streams and consumer config sections to look consistent and include all current fields.

- [streams](https://docs.nats.io/~/revisions/vU3k0NQVTL6lob566Fzu/nats-concepts/jetstream/streams#configuration)
- [consumers](https://docs.nats.io/~/revisions/vU3k0NQVTL6lob566Fzu/nats-concepts/jetstream/consumers#configuration)

File: nats-concepts/jetstream/consumers.md [hunk 1]
```diff
@@ -1,234 +1,94 @@
...
 
-The rate of message delivery in both cases is subject to `ReplayPolicy`. A `ReplayInstant` Consumer will receive all messages as fast as possible while a `ReplayOriginal` Consumer will receive messages at the rate they were received, which is great for replaying production traffic in staging.
+In addition to the choice of being push or pull, a consumer can also be **ephemeral** or **durable**. A durable consumer simply by having a *durable name* defined when creating the consumer initially, whereas an ephemeral does not. Durables and ephemeral behave exactly the same except that an ephemeral will be automatically *cleaned up* (deleted) after a period of inactivity, specifically when there are no subscriptions bound to the consumer. By default, durables will remain even when there are periods of inactivity (unless `InactiveThreshold` is set explicitly).
```
Comment (gcolliso) [inline nats-concepts/jetstream/consumers.md:7]: Sentence is missing a word "A durable consumer simply by having..."

File: nats-concepts/jetstream/consumers.md [hunk 2]
```diff
@@ -1,234 +1,94 @@
...
 
-In the orders example above we have 3 Consumers. The first two select a subset of the messages from the Stream by specifying a specific subject like `ORDERS.processed`. The Stream consumes `ORDERS.*` and this allows you to receive just what you need. The final Consumer receives all messages in a `push` fashion.
+For examples on how to configure consumers with your preferred NATS client, see [NATS by Example](https://natsbyexample.com).
```
Comment (gcolliso) [inline nats-concepts/jetstream/consumers.md:9]: If there is a specific example in the NBE, IMO the link should go directly to it so ppl don't have to hunt around.

File: nats-concepts/jetstream/consumers.md [hunk 3]
```diff
@@ -1,234 +1,94 @@
...
+- `DeliverLast` - When first consuming messages, the consumer will start receiving messages with the last message added to the stream, or the last message in the stream that matches the consumer's filter subject if defined.
+- `DeliverLastPerSubject` - When first consuming messages, start with the latest one for each filtered subject currently in the stream.
+- `DeliverNew` - When first consuming messages, the consumer will only start receiving messages that were created after the consumer was created.
+- `DeliverByStartSequence` - When first consuming messages, start at this particular message in the set. The consumer is required to specify `OptStartSeq`, the sequence number to start on. It will receive the closest available message moving forward in the sequence should the message specified have been removed based on the stream limit policy.
+- `DeliverByStartTime` - When first consuming messages, start with messages on or after this time. The consumer is required to specify `OptStartTime`, the time in the stream to start at. It will receive the closest available message on or after that time.
```
Comment (gcolliso) [inline nats-concepts/jetstream/consumers.md:61]: I know I'm the grammar police here but can we reword this one "The consumer is required to specify...., the time in the stream to start at." to '....the time at which the stream should start.' ?

File: nats-concepts/jetstream/streams.md [hunk 4]
```diff
@@ -12,34 +12,36 @@ Streams support various retention policies which define when messages in the str
 
 Streams support deduplication using a `Nats-Msg-Id` header and a sliding window within which to track duplicate messages. See the [Message Deduplication](../../using-nats/jetstream/model_deep_dive.md#message-deduplication) section.
 
+For examples on how to configure streams with your preferred NATS client, see [NATS by Example](https://natsbyexample.com).
```
Comment (gcolliso) [inline nats-concepts/jetstream/streams.md:15]: Same as above, if there is a specific NBE, the link should go there.

**Comment by bruth** [comment]
Thanks @gcolliso. Hopefully addressed all the feedback.

**Review by gcolliso — APPROVED** [review]
lgtm - thanks @bruth

File: nats-concepts/jetstream/consumers.md [hunk 5]
```diff
@@ -1,234 +1,91 @@
...
+| MaxDeliver | The maximum number of times a specific message delivery will be attempted. Applies to any message that is re-sent due to ack policy (i.e. due to a negative ack, or no ack sent by the client). | 2.2.0 | Yes |
+| ReplayPolicy | If the policy is `ReplayOriginal`, the messages in the stream will be pushed to the client at the same rate that they were originally received, simulating the original timing of messages. If the policy is `ReplayInstant` \(the default\), the messages will be pushed to the client as fast as possible while adhering to the Ack Policy, Max Ack Pending and the client's ability to consume those messages. | 2.2.0 | No |
+| Replicas | Sets the number of replicas for the consumer's state. By default, consumers inherit the number of replicas from the stream. | 2.8.3 | Yes |
```
Comment (jnmoyne) [inline nats-concepts/jetstream/consumers.md:31]: About consumer number of replicas maybe expand to say that the default value 0 means it inherits the number from the stream's

File: nats-concepts/jetstream/consumers.md [hunk 6]
```diff
@@ -1,234 +1,91 @@
...
+- `DeliverByStartTime` - When first consuming messages, start with messages on or after this time. The consumer is required to specify `OptStartTime` which defines this start time.
 
-You do not have to ack any messages, the server will assume ack on delivery.
+### MaxAckPending
 
-### AckAll
+The `MaxAckPending` capability provides one-to-many flow control and applies to both push and pull consumers. For push consumers, `MaxAckPending` is the _only_ form of flow control. However, for pull consumers because the delivery of the messages to the client application is client-driven (hence the 'pull') rather than server initiated (hence the 'push') there is an implicit one-to-one flow control with the subscribers (the maximum batch size of the Fetch calls). Therefore you should remember to set it to an appropriately high value (e.g. the default value of 20000), as it can otherwise place a limit on the horizontal scalability of the processing of the stream in high throughput situations.
```
Comment (jnmoyne) [inline nats-concepts/jetstream/consumers.md:63]: Default max ack pending value has changed and is now `1000` I believe

File: nats-concepts/jetstream/consumers.md [hunk 7]
```diff
@@ -1,234 +1,91 @@
...
-
-{% hint style="info" %}
-The consumer's description _can_ be adjusted after the initial creation of the consumer
-{% endhint %}
+| Field | Description | Version | Editable |
+| :--- | :--- | :--- | :--- |
+| DeliverSubject | The subject to deliver messages to. Note, setting this field implicitly decides whether the consumer is push or pull-based. With a deliver subject, the server will _push_ messages to client subscribed to this subject. | 2.2.0 | No |
+| DeliverGroup | The queue group name which, if specified, is then used to distribute the messages between the subscribers to the consumer. This is analogous to a [queue group][queue-group] in core NATS. | 2.2.0 | Yes |
+| FlowControl | Enables per-subscription flow control using a sliding-window protocol. This protocol relies on the server and client exchanging messages to regulate when and how many messages are pushed to the client. This flow control mechanism is in addition to the flow control imposed by `MaxAckPending` across all subscriptions bound to a consumer. | 2.2.0 | Yes |
```
Comment (jnmoyne) [inline nats-concepts/jetstream/consumers.md:86]: Maybe add 'one-to-one' somewhere to say that this FlowControl setting is for having individual flow control for each client subscribing from the consumer (while max ack pending is 'one-to-many' flow control)?

File: nats-concepts/jetstream/consumers.md [hunk 8]
```diff
@@ -1,234 +1,91 @@
...
+| IdleHeartbeat | If the idle heartbeat period is set, the server will regularly send a status message to the client (i.e. when the period has elapsed) while there are no new messages to send. This lets the client know that the JetStream service is still up and running, even when there is no activity on the stream. The message status header will have a code of 100. Unlike FlowControl, it will have no reply to address. It may have a description like "Idle Heartbeat". | 2.2.0 | Yes |
```
Comment (jnmoyne) [inline nats-concepts/jetstream/consumers.md:87]: Maybe say that this handling of the heartbeat is done by the client library, doesn't have to be done by the developper?

**Comment by bruth** [comment]
@jnmoyne Addressed the feedback thus far.

**Comment by jnmoyne** [comment]
LGTM

**Pull request merged**